test/expect-tests/dune_rules/cram_parsing_tests.ml

```diff
@@ -4,49 +4,130 @@ open Dune_rules.For_tests.Cram_exec.For_tests
 let () = Dune_tests_common.init ()
 
 let test content =
-  Lexing.from_string content
-  |> cram_stanzas
-  |> Dyn.list dyn_of_block
+  let lexbuf = Lexing.from_string ~with_positions:true content in
+  Stdlib.Lexing.set_filename lexbuf "test";
+  cram_stanzas lexbuf
+  |> Dyn.list (Dyn.pair Loc.to_dyn dyn_of_block)
   |> Dyn.pp
   |> List.singleton
   |> Dune_console.print
 ;;
 
 let%expect_test "simple single line command" =
   test "  $ echo 'Hello world'";
-  [%expect {| [ Command [ "echo 'Hello world'" ] ] |}];
+  [%expect
+    {|
+    [ ({ pos_fname = "test"
+       ; start = { pos_lnum = 1; pos_bol = 0; pos_cnum = 2 }
+       ; stop = { pos_lnum = 1; pos_bol = 0; pos_cnum = 22 }
+       },
+       Command [ "echo 'Hello world'" ])
+    ]
+    |}];
   test "  $ echo 'Hello world'\n";
-  [%expect {| [ Command [ "echo 'Hello world'" ] ] |}]
+  [%expect
```

Comment:
The way the tests are currently written makes it really hard to tell if this is doing the right thing. How about using the machinery in `Loc` to actually show where the location is pointing to?